components/post_edit_history/edited_post_item/edited_post_item.tsx

```diff
@@ -0,0 +1,240 @@
+                },
+            },
+        };
+
+        actions.openModal(infoToastModalData);
+    };
+
+    const handleRestore = async () => {
+        if (!originalPost || !post || originalPost.message === post.message) {
+            actions.closeRightHandSide();
+            return;
+        }
+
+        const updatedPost = {
+            message: post.message,
+            id: originalPost.id,
+            channel_id: originalPost.channel_id,
+        };
+
+        const result = await actions.editPost(updatedPost as Post);
+        if (result.data) {
+            actions.closeRightHandSide();
+            showInfoTooltip();
+        }
+    };
+
+    const handleUndo = async () => {
+        if (!originalPost) {
+            actions.closeRightHandSide();
+            return;
+        }
+
+        await actions.editPost(originalPost);
+    };
+
+    const currentVersionIndicator = isCurrent ? (
+        <div className='edit-post__current__indicator'>
+            {currentVersionText}
+        </div>
+    ) : undefined;
+
+    const profileSrc = imageURLForUser(post.user_id);
+
+    const overwriteName = post.props ? post.props.override_username : '';
+    const postHeader = (
+        <div className='edit-post__header'>
+            <span className='profile-icon'>
+                <Avatar
+                    size={'sm'}
+                    url={profileSrc}
+                    className={'avatar-post-preview'}
+                />
+            </span>
+            <div className={'edit-post__header__username'}>
+                <UserProfileComponent
+                    userId={post.user_id}
+                    hasMention={true}
+                    disablePopover={true}
+                    overwriteName={overwriteName}
+                />
+            </div>
+        </div>
+    );
+
+    const message = (
+        <PostBodyAdditionalContent post={post}>
+            <PostMessageContainer
+                post={post}
+                isRHS={true}
+                showPostEditedIndicator={false}
+            />
+        </PostBodyAdditionalContent>
+    );
+
+    const messageContainer = (
+        <>
+            {postHeader}
+            <div className='post__content' >
+                <div className='search-item-snippet post__body'>
+                    {message}
+                </div>
+            </div >
+        </>
+    );
+
+    const tooltip = (
+        <Tooltip
+            id='editPostRestoreTooltip'
+            className='hidden-xs'
+        >
+            {formattedHelpText}
+        </Tooltip>
+    );
+
+    const restoreButton = isCurrent ? undefined : (
+        <OverlayTrigger
+            trigger={['hover', 'focus']}
+            delayShow={Constants.OVERLAY_TIME_DELAY}
+            placement='bottom'
+            overlay={tooltip}
+        >
+            <IconButton
+                size={'sm'}
+                icon={'refresh'} // todo update with correct icon. Restore icon is recently added to compass
```

Comment:
Yes, that would be great.

Reply:
Let me know how to proceed with this.

Reply:
I'll create a followup to deal with this.

@michelengelen Is IconButton component exclusive to compass-components, or is there a compass-icons version of it? I might be confusing that with the various icon components that are generated for it

Reply:
the `IconButton` is indeed a compass-component thing. I was under the impression that it will just use whatever we have as version for compass-icons in the webapp, but apparently that does not work as expected (or I did have an old version locally and thats why it did not work, will cross-check that.

Reply:
Apparently it should work. @sinansonmez Would you mind updating this branch with current `master` and try to use the Restore icon here?